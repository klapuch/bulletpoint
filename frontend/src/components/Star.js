// @flow

import React from 'react';
import styled from 'styled-components';
import classNames from 'classnames';

const Resized = styled.span`
  font-size: 34px;
  color: #f2b01e;
  cursor: pointer;
`;

type Props = {|
  +active: boolean,
  +onClick: (boolean) => (void),
|};
type State = {|
  active: boolean,
|};
export default class Star extends React.Component<Props, State> {
  state = {
    active: false,
  };

  componentDidMount(): void {
    this.setState({ active: this.props.active });
  }

  mark = () => {
    if (!this.props.active) {
      this.setState((prevState: State) => ({ active: !prevState.active }));
    }
  };

  render() {
    const { active } = this.state;
    return (
      <Resized
        onFocus={this.mark}
        onMouseOver={this.mark}
        onBlur={this.mark}
        onMouseOut={this.mark}
        onClick={() => this.props.onClick(!this.props.active)}
        className={classNames('glyphicon', active ? 'glyphicon-star' : 'glyphicon-star-empty')}
        aria-hidden="true"
      />
    );
  }
}
