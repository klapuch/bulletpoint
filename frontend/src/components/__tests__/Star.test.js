import React from 'react';
import renderer from 'react-test-renderer';
import Star from '../Star';

test('renders active star', () => {
  const component = renderer.create(
    <Star active onClick={() => null} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders not active star', () => {
  const component = renderer.create(
    <Star active={false} onClick={() => null} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
