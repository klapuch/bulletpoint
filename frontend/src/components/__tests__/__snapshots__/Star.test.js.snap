// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active star 1`] = `
<span
  aria-hidden="true"
  className="glyphicon glyphicon-star sc-bdVaJa aAjEk"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
/>
`;

exports[`renders not active star 1`] = `
<span
  aria-hidden="true"
  className="glyphicon glyphicon-star-empty sc-bdVaJa aAjEk"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
/>
`;
